exporter/collector/googlemanagedprometheus/monitoredresource.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlemanagedprometheus
+
+import (
+	"go.opentelemetry.io/collector/model/pdata"
+	semconv "go.opentelemetry.io/collector/model/semconv/v1.8.0"
+	monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
+)
+
+func MapToPrometheusTarget(res pdata.Resource) *monitoredrespb.MonitoredResource {
+	attrs := res.Attributes()
+	// Prepend namespace if it exists to match what is specified in
+	// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#resource-attributes-1
+	job := getStringOrEmpty(attrs, semconv.AttributeServiceName)
+	serviceNamespace := getStringOrEmpty(attrs, semconv.AttributeServiceNamespace)
+	if serviceNamespace != "" {
+		job = serviceNamespace + "/" + job
+	}
+	location := getStringOrEmpty(attrs, semconv.AttributeCloudAvailabilityZone)
+	if location == "" {
```

Comment:
Can you add a TODO and link to the issue mentioning the resource detection logic fix? Note that's also a breaking change for the exporter once merged here.

Reply:
Nothing needs to be done in the exporter, which already has the correct behavior.  Once the fix is in the resource detector, it will work correctly without any additional changes here.

Reply:
Does that mean the GKE resource detector will use `cluster-location` instance attribute, determine whether the location is a zone or region (how feasible is this?), and set the zone/region attribute?

Reply:
Yes, that is exactly what it will do.  I have a [WIP detector](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/383/files#diff-67c35f4fb2417f61d45777e33b607496b52caba05000bc350d4b22b3a9b3ebc5R40-R50) that does this for SDK detection if you are curious.  It should be feasible to do.

Reply:
I see. Would it be more robust to add a new `location` resource attribute and use that? That said I don't know how difficult it is to do so.

Reply:
Resource attributes are defined by the otel semantic conventions: https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/resource/cloud.yaml#L45.  Right now, it defines zone + region.  I suspect it would be difficult to convince them to add a third convention, "location", which is the zone or region.

Reply:
I see. It seems possible to add this as a GKE-specific attribute, but maybe that's an overkill.